src/config/cloudtrail.json

```diff
@@ -0,0 +1,1006 @@
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBClusterSnapshot', &dt_meid_supporting_service_v2( 'Aurora', \"aws.arn\"), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 10,
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBClusterSnapshot', &format_arn('arn:{}:rds:{}:{}:cluster:{}', [partition, region, account_id, log_content_parsed.requestParameters.dBClusterIdentifier]), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBClusterSnapshot', &log_content_parsed.requestParameters.dBClusterIdentifier, &EmptyValue, @ )"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBInstance', &dt_meid_rds_v2( \"aws.arn\"), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 10,
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBInstance', &format_arn('arn:{}:rds:{}:{}:db:{}', [partition, region, account_id, log_content_parsed.responseElements.dBInstanceIdentifier]), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBInstance', &log_content_parsed.responseElements.dBInstanceIdentifier, &EmptyValue, @ )"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBInstanceReadReplica', &dt_meid_rds_v2( \"aws.arn\"), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 10,
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBInstanceReadReplica', &format_arn('arn:{}:rds:{}:{}:db:{}', [partition, region, account_id, log_content_parsed.responseElements.sourceDBInstanceIdentifier]), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBInstanceReadReplica', &log_content_parsed.responseElements.sourceDBInstanceIdentifier, &EmptyValue, @ )"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBSnapshot', &dt_meid_rds_v2( \"aws.arn\"), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 10,
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBSnapshot', &format_arn('arn:{}:rds:{}:{}:db:{}', [partition, region, account_id, log_content_parsed.requestParameters.dBInstanceIdentifier]), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'CreateDBSnapshot', &log_content_parsed.requestParameters.dBInstanceIdentifier, &EmptyValue, @ )"
```

Comment:
Well, I don't know how much performance impact we have with those ifs. But they are clearly exclusive. Switch statement might reduce number of comparisons significantly, i.e. once you match one eventName (optimistically one in first half ;) ) then you don't need to compare rest of them. 
I am not sure If we want to change it anyhow, just pointing out that this is not optimal from performance point of view

Reply:
Sound like a good optimization, then we'd only need a few additional if-comparisons. As this would be a pretty significant change to the JSON schema, I'd do it after the simple profiling you proposed in another comment to check if it's worth the effort. WDYT @dt-be ?

Reply:
There are a couple pretty invasive things that could be done here. I would like to go in any of those directions only based on profiling results.